core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/chain/StopStartDownscaleFlowEventChainFactory.java

```diff
@@ -42,24 +36,22 @@ public String initEvent() {
     public FlowTriggerEventQueue createFlowTriggerEventQueue(ClusterAndStackDownscaleTriggerEvent event) {
 
         StackView stackView = stackService.getViewByIdWithoutAuth(event.getResourceId());
-        ClusterView clusterView = stackView.getClusterView();
-        HostGroup hostGroup = hostGroupService.getByClusterIdAndName(clusterView.getId(), event.getHostGroupName())
-                .orElseThrow(NotFoundException.notFound("hostgroup", event.getHostGroupName()));
-
-        StopStartDownscaleTriggerEvent te = new StopStartDownscaleTriggerEvent(
-                StopStartDownscaleEvent.STOPSTART_DOWNSCALE_TRIGGER_EVENT.event(),
-                stackView.getId(),
-                hostGroup.getName(),
-                Sets.newHashSet(event.getPrivateIds()),
-                event.getClusterManagerType()
-        );
+        Map<String, Set<Long>> hostGroupsWithPrivateIds = event.getHostGroupsWithPrivateIds();
 
         Queue<Selectable> flowEventChain = new ConcurrentLinkedQueue<>();
 
         // TODO CB-14929: Is a stack sync really required here. What does it do ? (As of now it also serves to accept the event)
         addStackSyncTriggerEvent(event, flowEventChain);
 
-        flowEventChain.add(te);
+        for (Map.Entry<String, Set<Long>> hostGroupWithPrivateIds : hostGroupsWithPrivateIds.entrySet()) {
```

Comment:
This isn't meant to handle multiple hostGroups right now. Better to add a precondition check, and fail?

Reply:
Agree, I add precondition check.